Source/Core/DolphinQt2/MainWindow.cpp

```diff
@@ -272,18 +281,46 @@ void MainWindow::Pause()
 
 bool MainWindow::Stop()
 {
+  if (!Core::IsRunning())
+    return true;
+
   bool stop = true;
+  Core::State state = Core::GetState();
+  // Set to false when Netplay is running as a CPU thread
+  bool pause = true;
+
   if (Settings::Instance().GetConfirmStop())
   {
-    // We could pause the game here and resume it if they say no.
+    if (pause)
+      Core::SetState(Core::State::Paused);
+
     QMessageBox::StandardButton confirm;
-    confirm = QMessageBox::question(m_render_widget, tr("Confirm"), tr("Stop emulation?"));
+    confirm = QMessageBox::question(m_render_widget, tr("Confirm"),
+                                    m_stop_requested ?
+                                        tr("A shutdown is already in progress. Unsaved data "
+                                           "may be lost if you stop the current emulation "
+                                           "before it completes. Force stop?") :
+                                        tr("Do you want to stop the current emulation?"));
     stop = (confirm == QMessageBox::Yes);
+
+    if (pause)
+      Core::SetState(state);
   }
 
+  // TODO: Add Movie shutdown
+  // TODO: Add Debugger shutdown
+
   if (stop)
```

Comment:
This is starting to get a bit long for nested ifs. Consider returning early here.